### ISSUE TITLE
_evalUrl isn't Content-Security-Policy (CSP) compatible

### ISSUE THREAD

**Issue by gavinwahl** [jquery/jquery#2012]
When using .html() with an HTML string that contains a script tag, it fetches the script source with XHR and evals it. I'd expect it to instead insert a script tag into the dom and let the browser handle it. With a CSP policy, inserting a script tag referencing the js file works but evaling its source doesn't.

# Test case

```html
<!DOCTYPE html>
<html>
<head>
  <script src="https://code.jquery.com/jquery-2.1.1.min.js"></script>
  <!-- nonce-foo is just to allow the inline script below with the test case -->
  <meta http-equiv="Content-Security-Policy" content="script-src 'self' 'nonce-foo' code.jquery.com; default-src 'self'">
</head>
<body>
  <script nonce="foo">
    $(document.body).html('<script src="/foo.js"><' + '/script>')
  </script>
</body>
</html>
```

I ran this test case using Chrome 39:

# Observed behavior

jQuery attempts to eval the source of `/foo.js` and causes a CSP violation: 

> Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self' 'nonce-foo' code.jquery.com".

# Desired behavior

A script node with a source attribute of `/foo.js` is inserted into the DOM. The browser executes it because 'self' is allowed script-src by the CSP policy. I know that this is possible because if the script has an external src (like `http://example.com/foo.js`), the script tag gets inserted instead of the xhr+eval thing. (Presumably because the same-origin policy won't allow XHR with example.com)

**Comment by mzgol** [comment]
This has been fixed in https://github.com/jquery/jquery/pull/1449, you just need to wait for jQuery 3.0.0.

In the meantime, could you check https://code.jquery.com/jquery-git.js to make sure it works fine with your scenario?

Closing for now, I'll reopen if it still doesn't work in your scenario.

**Comment by mzgol** [comment]
Thanks for the report! This has been fixed in https://github.com/jquery/jquery/pull/1449, you just need to wait for jQuery 3.0.0.

In the meantime, could you check https://code.jquery.com/jquery-git.js to make sure it works fine with your scenario?

Closing for now, I'll reopen if it still doesn't work.

**Comment by gavinwahl** [comment]
jquery-git doesn't work. The only difference is that an inline script is used instead of eval. ie, `<script>...xhr'd source code here...</script>` is used, not `<script src="..script url here..."></script>`

**Comment by dmethvin** [comment]
Actually the ticket that would fix this is #1895, which we do plan to land for 3.0. It's a big breaking change because right now we run scripts synchronously if they come from the same domain, so we're not sure the effect it might have on the world.

**Comment by gavinwahl** [comment]
I wrote some code (probably hopelessly naive and will break something else) that makes it work:

```patch
diff --git a/src/manipulation/_evalUrl.js b/src/manipulation/_evalUrl.js
index 0fdacbc..1559804 100644
--- a/src/manipulation/_evalUrl.js
+++ b/src/manipulation/_evalUrl.js
@@ -3,15 +3,9 @@ define([
 ], function( jQuery ) {
 
 jQuery._evalUrl = function( url ) {
-       return jQuery.ajax({
-               url: url,
-               type: "GET",
-               dataType: "script",
-               cache: true,
-               async: false,
-               global: false,
-               "throws": true
-       });
+       var s = document.createElement('script');
+       s.src = url;
+       document.body.appendChild(s);
 };
 
 return jQuery._evalUrl;
```

**Comment by mzgol** [comment]
Ah, right. @dmethvin, I guess this can stay closed since there's already #1895 for that?

**Comment by dmethvin** [comment]
I think we can track it there, yeah.
